python3Packages.gdtoolkit: Use lark_0_8_0 (split from python3Packages.lark) and enable tests

hesiod opened the pull request:
###### Description of changes

Based on my findings in https://github.com/NixOS/nixpkgs/pull/216280#issuecomment-1429521866 (gdtoolkit is broken with lark 1.1.5):
* ~~Split `python3Packages.lark` into `python3Packages.lark_0_8_0` and `python3Packages.lark_1_1_5` (`python3Packages.lark` points to `python3Packages.lark_1_1_5`)~~
* ~~Use `python3Packages.lark_0_8_0` for `python3Packages.gdtoolkit`~~

Edit: I've edited the PR based on https://github.com/NixOS/nixpkgs/pull/216329#issuecomment-1430165592:
* Move `gdtoolkit` from `pkgs/development/python-modules` to `pkgs/development/tools/gdtoolkit` (also, remove gdtoolkit from `pkgs/top-level/python-packages.nix` and add it to `pkgs/top-level/all-packages.nix`)
* Run gdtoolkit tests

PR is based on #216280.
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

hesiod committed:
python3Packages.gdtoolkit: Enable tests

hesiod committed:
python3Packages.gdtoolkit: Add maintainer tmarkus

hesiod commented:
Marked as draft until #216280 is merged, but this PR is ready for review.

FRidh reviewed (changes requested):
Multiple versions of the same runtime package are not allowed in the Python packages set. I see it was already the case before, that was not OK either. We need to choose *one* version.

hesiod commented:
@FRidh OK, so should we downgrade `lark` to `0.8.0` (or perhaps `0.8.9` if that works with gdtoolkit)?
As mentioned, gdtoolkit is definitely broken with newer lark versions.
Are there any alternatives?

FRidh commented:
Looks like `gdtoolkit` is just a set of executables. It can be moved  out of `python-packages.nix`, use `buildPythonApplication`, and then its own version of `lark`.

hesiod commented:
@FRidh Good point. Not sure why this was missed in the original PR (#160040). I'll look into this.

hesiod committed:
python3Packages.gdtoolkit: Move to pkgs/development/tools

gdtoolkit is not a library. Move to pkgs/development/tools and use
buildPythonApplication.

Fixes #216243.

hesiod commented:
@FRidh I've implemented your suggestion. See https://github.com/NixOS/nixpkgs/pull/216329/commits/5c8a28b9276dff5f6d1ba9230108c55851e61c21
Please let me know if that works for you.

kirillrdy commented on pkgs/top-level/all-packages.nix:
```diff
@@ -7581,6 +7581,8 @@ with pkgs;
 
   gdmap = callPackage ../tools/system/gdmap { };
 
+  gdtoolkit = callPackage ../development/tools/gdtoolkit { };
```
suggestion: add gdtoolkit to `python-aliases.nix` so that people know that it got moved

hesiod replied:
Sure, will do!

hesiod replied:
fixed in rebased commits

shiryel reviewed (approved):
`nix-review pr 216329` building and running as expected.
